File: eventual-consistency-support-core/src/main/java/org/xyattic/eventual/consistency/support/core/persistence/ProviderPersistence.java
package org.xyattic.eventual.consistency.support.core.persistence;

import org.xyattic.eventual.consistency.support.core.provider.PendingMessage;
import org.xyattic.eventual.consistency.support.core.provider.enums.PendingMessageStatus;

import java.util.Date;
import java.util.List;

/**
 * @author wangxing
 * @create 2020/4/14
 */
public interface ProviderPersistence {

    void save(List<PendingMessage> pendingMessages);

    void changePendingMessageStatus(String id, PendingMessageStatus status);

    List<PendingMessage> getPendingMessages(Date timeBefore);

}

File: eventual-consistency-support-core/src/main/java/org/xyattic/eventual/consistency/support/core/persistence/ProviderPersistence.kt
package org.xyattic.eventual.consistency.support.core.persistence

import org.xyattic.eventual.consistency.support.core.provider.PendingMessage
import org.xyattic.eventual.consistency.support.core.provider.enums.PendingMessageStatus
import java.util.*

/**
 * @author wangxing
 * @create 2020/4/14
 */
interface ProviderPersistence {

    fun save(pendingMessage: PendingMessage)

    fun changePendingMessageStatus(id: String, status: PendingMessageStatus, sendTime: Date)

    fun sendSuccess(id: String, messageId: String)

    fun sendFailed(id: String)

    fun getPendingMessages(timeBefore: Date): List<PendingMessage>

}